src/halite/features/cache.cr

```diff
@@ -0,0 +1,109 @@
+require "json"
+require "digest"
+require "file_utils"
+
+module Halite::Features
+  # Cache feature use for caching HTTP response to local storage to speed up in developing stage.
+  #
+  # It has the following options:
+  #
+  # - `path`: The path of cache, default is "cache/"
+  # - `expires`: The expires time of cache, default is nerver expires.
+  # - `debug`: The debug mode of cache, default is `true`
+  #
+  # With debug mode, cached response it always included some headers information:
+  #
+  # - `X-Cached-Key`: Cache key with verb, uri and body
+  # - `X-Cached-At`:  Cache created time
+  # - `X-Cached-Expires-At`: Cache expired time
+  # - `X-Cached-By`: Always return "Halite"
+  #
+  # ```
+  # Halite.use("cache").get "http://httpbin.org/anything"         # request a HTTP
+  # r = Halite.use("cache").get "http://httpbin.org/anything"     # request from local storage
+  # r.headers # => {..., "X-Cached-At" => "2018-08-30 10:41:14 UTC", "X-Cached-By" => "Halite", "X-Cached-Expires-At" => "2018-08-30 10:41:19 UTC", "X-Cached-Key" => "2bb155e6c8c47627da3d91834eb4249a"}}
+  # ```
+  class Cache < Feature
+    DEFAULT_PATH = "cache/"
```

Comment:
It's not a good idea to create a directory in the root directory, at least create in `/tmp`
Maybe a better approach would be:
1. On a new `Cache` object creation, a temporary directory will be created with https://crystal-lang.org/api/master/Tempfile.html (`Tempfile.new("halite-cache").path`)
2. Add a `finalize` method that will delete the directory when the Object is destructed (`#delete`)

Reply:
This commit just a simple test, it was not following my spec. May be push it early.